platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/AbstractPKIHandler.java

```diff
@@ -68,9 +90,63 @@ public HandlerResult getNormalizedToken(ServletRequest request, ServletResponse
                 (X509Certificate[]) httpRequest
                         .getAttribute("javax.servlet.request.X509Certificate"));
 
-        if (token != null) {
+        X509Certificate[] certs = (X509Certificate[]) request
+                .getAttribute("javax.servlet.request.X509Certificate");
+
+        HttpServletResponse httpResponse = response instanceof HttpServletResponse ?
+                (HttpServletResponse) response :
+                null;
+
+        // Somehow the httpResponse was null, return no action and try to process with other handlers
+        if (httpResponse == null) {
+            LOGGER.error("Somehow the httpResponse was null");
```

Comment:
Lets give a better error message than this. Maybe something like "HTTP Response was null for request: <path they are trying to hit>"